app/browser/reducers/tabsReducer.js

```diff
@@ -337,6 +337,31 @@ const tabsReducer = (state, action, immutableAction) => {
         }
       }
       break
+    case appConstants.APP_WINDOW_READY: {
+      console.log(action)
+      if (!action.getIn(['createProperties', 'windowId'])) {
+        const senderWindowId = action.getIn(['senderWindowId'])
+        if (senderWindowId) {
+          action = action.setIn(['createProperties', 'windowId'], senderWindowId)
+        }
+      }
+
+      const welcomeScreenProperties = {
+        'url': 'about:welcome',
+        'windowId': action.getIn(['createProperties', 'windowId'])
+      }
+
+      // tests should not be aware of the welcome screen
+      const shouldShowWelcomeScreen = process.env.NODE_ENV !== 'test' &&
```

Comment:
we should probably have a way to override this so we can test it. Maybe it would be better to set the defaultState for about.welcome.showOnLoad in sessionStore.js to `true` except for production and then we set it to `false` when we want to test?